### PR TITLE
feat: extract generation process from prompt window

### PR DESCRIPTION
**Pull request by TheLLspectre**
Extract all the generation process from the prompt window.
Add creationMode Class to add and manage all mode from webApp (ControlNet, IpAdapter, etc...)
Plug prompt window UI on it.
Add following modes:
- Text To Image
- Image To Image
- ControlNet
- In Painting
- Ip Adapter
- Reference Only
- Image To Image + Control Net
- Image To Image + Ip Adapter
- Control Net + Ip Adapter

- closes: #166 #167

### CODE REVIEW

File: package/Editor/PromptWindow/Views/ImageSettingsView.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ private void RenderImageSettingsSection(bool shouldAutoGenerateSeed)
             EditorGUILayout.BeginHorizontal();
             {
                 GUILayout.Label("Scheduler: ");
-                schedulerIndex = EditorGUILayout.Popup(schedulerIndex, schedulerOptions);
+                pusher.schedulerSelected = EditorGUILayout.Popup(pusher.schedulerSelected, pusher.SchedulerOptions);
```
[the remark this answers is not in the document]
Comment (TheLLspectre) [reply package/Editor/PromptWindow/Views/ImageSettingsView.cs:54]: I choose promptPusher !

File: package/Editor/_Services/PromptFetcher.cs [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
 using Newtonsoft.Json;
-using Unity.EditorCoroutines.Editor;
 using UnityEngine;
 
 namespace Scenario.Editor
 {
+    /// <summary>
+    /// PromptFetcher Class manage all API request (post inference generation, get inferences generated)
+    /// </summary>
     public class PromptFetcher
```
[the remark this answers is not in the document]
Comment (TheLLspectre) [reply package/Editor/_Services/PromptFetcher.cs:12]: I think about something with API but InferenceManager should be nice

**Comment by TheLLspectre** [comment]
@Morgan-6Freedom It's done
Something wrong with the pull request's jobs ??

**Comment by TheLLspectre** [comment]
Also forget to rename the file ! May be this can cause breaking change ?

**Review by Morgan-6Freedom — DISMISSED** [review]

**Comment by Morgan-6Freedom** [comment]
image

**Comment by Morgan-6Freedom** [comment]
waiting for @qvaleroo test before merging 👍

**Comment by qvaleroo** [comment]
https://github.com/scenario-labs/Scenario-Unity/assets/95273550/50e13efe-9b2e-4b1d-9152-97cfc8ce58c1

Set Image as reference seems broken.

**Comment by qvaleroo** [comment]
Issue requestion remove bg of my ref image:
image

here is my asset:
asset_X9vCdpsMi8hiKHS4of7y8LVD_pixar_upscale_1711437167

**Comment by qvaleroo** [comment]
`ControlNet` and `Inpaint`
image

**Comment by qvaleroo** [comment]
I don't get how the inpaint is working here? I think you are not uploading the mask because when I click on generate then I have this error:
image


https://github.com/scenario-labs/Scenario-Unity/assets/95273550/739fe2a7-0602-44a6-95d9-25d016038d12

**Comment by qvaleroo** [comment]
Reference Only:
- Style fidelity should be name Influence like others (value to send is Influence / 100)
- Checkboxes not aligned

image

**Comment by qvaleroo** [comment]
No space -> ControlNet
image

**Comment by qvaleroo** [comment]
image
`IP Adapter`

**Comment by TheLLspectre** [comment]
Draging & droping image not working correctly.

**Comment by TheLLspectre** [comment]
> Issue when requesting remove bg of my ref image, or sending my generation with a ref image image
> 
> here is my asset: asset_X9vCdpsMi8hiKHS4of7y8LVD_pixar_upscale_1711437167

So on dev with this image I can't remove BG too. But with other I can, Request is too long message is coming from the server side.

**Comment by qvaleroo** [comment]
Just need to rename `InPainting`for `Inpaint` and we are ok :D

**Review by qvaleroo — APPROVED** [review]

**Pull request merged**